.github/workflows/build-and-test.yml

```diff
@@ -433,3 +433,115 @@ jobs:
           choco uninstall -y splunk-otel-collector | Tee-Object -file .\dist\uninstallation_logs.log
           if (!((Get-CimInstance -ClassName win32_service -Filter "Name = 'splunk-otel-collector'" | Select Name, State).State -Eq "Running")) { write-host "splunk-otel-collector has been successfully uninstalled and service is not running." } else { throw "Failed to uninstall splunk-otel-collector chocolatey package." }
           if (!((Get-CimInstance -ClassName win32_service -Filter "Name = 'fluentdwinsvc'" | Select Name, State).State -Eq "Running")) { write-host "fluentdwinsvc has been successfully uninstalled and service is not running." } else { throw "Failed to uninstall fluentdwinsvc chocolatey package." }
+
+  docker-otelcol:
+    name: docker-otelcol
+    runs-on: ubuntu-latest
+    needs: [cross-compile]
+    strategy:
+      matrix:
+        ARCH: [ "amd64", "arm64" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+            ~/go/bin
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Set up QEMU
+        if: ${{ matrix.ARCH == 'arm64'}}
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: arm64
+
+      - name: Downloading binaries-linux_${{ matrix.ARCH }}
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries-linux_${{ matrix.ARCH }}
+          path: ./bin
+      
+      - name: Build ${{ matrix.ARCH }} docker image
+        run: |
+          make docker-otelcol SKIP_COMPILE=true ARCH=${{ matrix.ARCH }}
+
+      - name: Run docker image
+        run: |
+          # ensure that the arch in the image manifest is correct
+          [ "$( docker inspect --format='{{.Architecture}}' otelcol:${{ matrix.ARCH }} )" = "${{ matrix.ARCH }}" ] || exit 1
+          docker run --platform linux/${{ matrix.ARCH }} -d -e SPLUNK_ACCESS_TOKEN=12345 -e SPLUNK_REALM=fake-realm --name otelcol otelcol:${{ matrix.ARCH }}
+          sleep 10
+          if [ -z "$( docker ps --filter=status=running --filter=name=otelcol -q )" ]; then
+            docker logs otelcol
+            echo "Failing job execution: fail to start otelcol docker container in 10 seconds."
+            exit 1
+          fi
+      
+      - name: Save docker image
```

Comment:
Let's skip saving and archiving the image to save time.

Reply:
Removed save docker image step for the GitHub Action workflow now.